src/list/index.tsx

```diff
@@ -153,10 +156,44 @@ export const ListItem = listItemFactory(function ListItem({
 		!disabled && !active && onRequestActive();
 	}
 
-	const [firstChild, ...otherChildren] = children();
-	const { leading = undefined, primary, trailing = undefined } = isRenderResult(firstChild)
-		? { primary: [firstChild, ...otherChildren] }
-		: firstChild;
+	const [firstChild] = children();
+	let listContents: RenderResult;
+
+	if (isRenderResult(firstChild)) {
+		listContents = firstChild;
+	} else {
+		const { leading = undefined, primary, trailing = undefined } = firstChild;
+		listContents = (
+			<virtual>
+				{leading ? <span classes={themedCss.leading}>{leading}</span> : undefined}
+				<span classes={themedCss.primary}>{primary}</span>
+				{trailing ? <span classes={themedCss.trailing}>{trailing}</span> : undefined}
+				{draggable && !trailing && (
+					<Icon
+						type="barsIcon"
+						classes={{ '@dojo/widgets/icon': { icon: [themedCss.dragIcon] } }}
+						theme={themeProp}
+						variant={variant}
+					/>
+				)}
+			</virtual>
+		);
+	}
+
+	let paddingClass: string | undefined;
+	switch (padding) {
+		case 'small':
+			paddingClass = themedCss.smallPadding;
+			break;
+		case 'medium':
```

Comment:
Should the options be 'small' and 'large'?

Reply:
could always add a large too, but that's fine by me